indexer/cli/cli.go

```diff
@@ -32,8 +37,15 @@ func runIndexer(ctx *cli.Context) error {
 		return err
 	}
 
-	cfg.Logger = logger
-	indexer, err := indexer.NewIndexer(cfg)
+	logger = log.NewLogger(cfg.Logger)
+
+	db, err := database.NewDB(cfg.DB)
+
+	if err != nil {
+		return err
+	}
+
+	indexer, err := indexer.NewIndexer(cfg.Chain, cfg.RPCs, db, logger)
```

Comment:
Being explicit with config instead of passing in the entire thing

Reply:
nit: logger being the first argument

Reply:
```suggestion
    // fixme logger should be first to stay consistent
	indexer, err := indexer.NewIndexer(cfg.Chain, cfg.RPCs, db, logger)
```
Leaving cleanup phase comment to fix since I bet I made this mistake elsewhere too